KubeArmor/core/crioHandler.go

```diff
@@ -312,17 +312,16 @@ func (dm *KubeArmorDaemon) MonitorCrioEvents() {
 
 	dm.Logger.Print("Started to monitor CRI-O events")
 
+	containers, err := Crio.GetCrioContainers()
```

Comment:
@Ankurk99 I think if we move this here, we would detect containers only once when kubearmor starts. After that, we won't get any new containers detected and added to the monitor. Also, same would happen with containerd too.

Reply:
:thinking: What my understanding is that we get all the present crio containers at the beginning and then whenever there is a new container added we are already calling UpdateContainer with start action.
https://github.com/kubearmor/KubeArmor/blob/main/KubeArmor/core/crioHandler.go#L340
Does that makes sense?

Reply:
I think currently [UpdateCrioContainer()](https://github.com/kubearmor/KubeArmor/blob/a28c6fbd124c306a4fcc481913edc171fcc6e860/KubeArmor/core/crioHandler.go#L199) just takes a container ID and gets info about that particular container from the CRI-API/Containerd API.
And the only function which gets all currently present containers is [GetCrioContainer()](https://github.com/kubearmor/KubeArmor/blob/a28c6fbd124c306a4fcc481913edc171fcc6e860/KubeArmor/core/crioHandler.go#L152). It will return all present containers and then we compare it with the container map in [CrioHandler](https://github.com/kubearmor/KubeArmor/blob/a28c6fbd124c306a4fcc481913edc171fcc6e860/KubeArmor/core/crioHandler.go#L32).
If both match, we didn't have any new containers and we continue, if the size of the returned map is greater, we call UpdateCrioContainer() to add it to this map .